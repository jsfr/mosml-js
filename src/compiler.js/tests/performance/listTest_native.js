"use strict"
var ctimeBegin$1 = function($_var1) {
    return console.time($_var1);
};
var ctimeEnd$2 = function($_var5) {
    return console.timeEnd($_var5);
};
var repeat$3 = function($_var9) {
    return function($_var10) {
        return function($_var11) {
            var $_var13 = function($_var14) {
                return (function() {
                    switch ($_var14) {
                        case 0:
                            return $_var10($_var11)
                        default:
                            return ($_var10($_var11), $_var13($_var14 - 1))
                    }
                }());
            };
            return $_var13($_var9 - 1);
        };
    };
};
var timerep$4 = function($_var35) {
    return function($_var36) {
        return function($_var37) {
            return repeat$3($_var35)(function($_var41) {
                return (ctimeBegin$1($_var37), ($_var36($_var41), ctimeEnd$2($_var37)));
            });
        };
    };
};

var listPrepend$5 = function($_var54) {
    var i = 0;
    var listRef = [];
    while (i++ < $_var54) {
        listRef.unshift(1);
    }
};
var listAppend$10 = function($_var54) {
    var i = 0;
    var listRef = [];
    while (i++ < $_var54) {
        listRef.push(1);
    }
};

var it$6 = timerep$4(10)(listPrepend$5)("listPrepend")(100000.0);
var it$12 = timerep$4(10)(listAppend$10)("listAppend")(1000.0);
