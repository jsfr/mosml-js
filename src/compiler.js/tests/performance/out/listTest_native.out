==============================JS BEGIN==============================
listPrepend: 1528ms
listPrepend: 1490ms
listPrepend: 1518ms
listPrepend: 1514ms
listPrepend: 1518ms
listPrepend: 1462ms
listPrepend: 1522ms
listPrepend: 1555ms
listPrepend: 1543ms
listPrepend: 1453ms
listPrepend: 1522ms
listPrepend: 1470ms
listPrepend: 1520ms
listPrepend: 1444ms
listPrepend: 1488ms
listPrepend: 1537ms
listPrepend: 1494ms
listPrepend: 1463ms
listPrepend: 1477ms
listPrepend: 1499ms
listPrepend: 1545ms
listPrepend: 1447ms
listPrepend: 1505ms
listPrepend: 1471ms
listPrepend: 1490ms
listPrepend: 1474ms
listPrepend: 1507ms
listPrepend: 1525ms
listPrepend: 1497ms
listPrepend: 1449ms
listPrepend: 1492ms
listPrepend: 1475ms
listPrepend: 1505ms
listPrepend: 1447ms
listPrepend: 1502ms
listPrepend: 1535ms
listPrepend: 1496ms
listPrepend: 1459ms
listPrepend: 1479ms
listPrepend: 1478ms
listPrepend: 1513ms
listPrepend: 1459ms
listPrepend: 1498ms
listPrepend: 1484ms
listPrepend: 1473ms
listPrepend: 1541ms
listPrepend: 1490ms
listPrepend: 1515ms
listPrepend: 1499ms
listPrepend: 1473ms
listPrepend: 1535ms
listPrepend: 1500ms
listPrepend: 1497ms
listPrepend: 1459ms
listPrepend: 1478ms
listPrepend: 1532ms
listPrepend: 1488ms
listPrepend: 1497ms
listPrepend: 1464ms
listPrepend: 1489ms
listPrepend: 1490ms
listPrepend: 1495ms
listPrepend: 1482ms
listPrepend: 1509ms
listPrepend: 1446ms
listPrepend: 1558ms
listPrepend: 1500ms
listPrepend: 1514ms
listPrepend: 1471ms
listPrepend: 1519ms
listPrepend: 1520ms
listPrepend: 1482ms
listPrepend: 1476ms
listPrepend: 1484ms
listPrepend: 1500ms
listPrepend: 1481ms
listPrepend: 1446ms
listPrepend: 1530ms
listPrepend: 1475ms
listPrepend: 1492ms
listPrepend: 1446ms
listPrepend: 1498ms
listPrepend: 1513ms
listPrepend: 1487ms
listPrepend: 1445ms
listPrepend: 1480ms
listPrepend: 1520ms
listPrepend: 1525ms
listPrepend: 1444ms
listPrepend: 1483ms
listPrepend: 1477ms
listPrepend: 1485ms
listPrepend: 1478ms
listPrepend: 1467ms
listPrepend: 1491ms
listPrepend: 1484ms
listPrepend: 1474ms
listPrepend: 1501ms
listPrepend: 1488ms
listPrepend: 1467ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 1ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
listAppend: 0ms
==============================JS END================================
